### ISSUE TITLE
Fix Request: AvoidUnguardedMutableInheretedFieldsInSharedObjects for for-each-loop variable

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#509]
false positive:
```
   private void renameOutgoingLinks(TenantId tenantId, RuleNodeId ruleNodeId, String oldLabel, String newLabel) {
        List<EntityRelation> relations = ruleChainService.getRuleNodeRelations(tenantId, ruleNodeId);
        for (EntityRelation relation : relations) {
            if (relation.getType().equals(oldLabel)) {
                relationService.deleteRelation(tenantId, relation);
                relation.setType(newLabel);  //  <- false positive
                relationService.saveRelation(tenantId, relation);
            }
        }
    }
```
